File: src/main/java/it/contrader/dto/BillDTO.java
package it.contrader.dto;

import java.sql.Date;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="id")

public class BillDTO {

	private Long id;

	private Long numberbill;
	
	private Date date;
	
	private String customer;
	
	private Float price;
	
	private LocationDTO location;


}

File: src/dto/billdto.ts
import { LocationDTO } from './locationdto';

export class BillDTO {

    id: number;
 
    customer: string;
 
    date: Date;
 
    location: LocationDTO = new LocationDTO();
 
    position: string;
 
    numberbill: number;
 
    price: number;
 
 }